chain/chain/src/chain.rs

```diff
@@ -959,15 +930,6 @@ impl Chain {
                 return Err(Error::InvalidBlockMerkleRoot);
             }
 
-            // Check that challenges root is empty to ensure later that block doesn't contain challenges.
-            // TODO (#2445): Enable challenges when they are working correctly.
-            if header.challenges_root() != &MerkleHash::default() {
-                return Err(Error::InvalidChallengeRoot);
-            }
-            if !header.challenges_result().is_empty() {
-                return Err(Error::InvalidChallenge);
-            }
```

Comment:
I was kinda waiting for this XD
We still need to check whether these fields are empty, even if they are deprecated, because malicious actor can fill them with some garbage. There is no obvious impact, BP just can bloat block sizes without clear benefit, but still seems cleaner to ensure that these fields are empty. Same for block header.
Maybe there is some borsh macro to enforce that. We also need to check that we don't change the serialisation.

Reply:
Good spot!  Agreed.  I introduced a new function to carry out this check.  PTAL.

I think longer term, we should introduce a `ValidatedBlockHeader` struct that is not part of the schema and that is validated.  I think it will simplify a lot of code and ensuring guarantees.

Reply:
Note the new check is in the top of the function.